File: Python Project/Ders.py
class Ders:
    def _init_(self, ders_adi, ders_kodu, donem, secmeli, ders_sorumlusu, kredi):
        self.ders_adi = ders_adi
        self.ders_kodu = ders_kodu
        self.donem = donem
        self.secmeli = secmeli
        self.ders_sorumlusu = ders_sorumlusu
        self.kredi = kredi
        self.ogrenci_listesi = []

    def not_hesapla(self, *notlar):
        raise NotImplementedError("Bu metod alt siniflar tarafindan uygulanmalidir.")

    def derse_yazilim(self, ogrenci):
        self.ogrenci_listesi.append(ogrenci)

    def ogrenci_kayitli_mi(self, numara):
        return any(ogrenci.numara == numara for ogrenci in self.ogrenci_listesi)

    def listeyi_yazdir(self):
        print(f"{self.ders_adi} Öğrenci Listesi:")
        for ogrenci in self.ogrenci_listesi:
            print(ogrenci.get_ad_soyad())

    def harf_notu(self, ortalama):
        if ortalama >= 90:
            return "AA"
        elif ortalama >= 80:
            return "BA"
        elif ortalama >= 70:
            return "BB"
        elif ortalama >= 60:
            return "CB"
        elif ortalama >= 50:
            return "CC"
        elif ortalama >= 40:
            return "DC"
        elif ortalama >= 30:
            return "DD"
        else:
            return "FF"

File: C# Project/ConsoleApp1/Ders.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    using System;
using System.Collections.Generic;
    using ConsoleApp1.obj;

    public abstract class Ders
{
    public string dersAdi;
    protected string dersKodu;
    protected string donem;
    protected bool secmeli;
    public int kredi;
    protected Akademisyen dersSorumlusu;
    private List<Ogrenci> ogrenciListesi;

    public Ders(string dersAdi, string dersKodu, string donem, Akademisyen dersSorumlusu, int kredi)
    {
        this.dersAdi = dersAdi;
        this.dersKodu = dersKodu;
        this.donem = donem;
        this.dersSorumlusu = dersSorumlusu;
        this.kredi = kredi;
        ogrenciListesi = new List<Ogrenci>();
    }

    public abstract string NotHesapla(params double[] notlar);

    public void DerseYazilim(Ogrenci ogrenci)
    {
        ogrenciListesi.Add(ogrenci);
    }

    public bool OgrenciKayitliMi(int numara)
    {
        foreach (Ogrenci ogrenci in ogrenciListesi)
        {
            if (ogrenci.getNumara() == numara)
            {
                return true;
            }
        }
        return false;
    }

    public void ListeyiYazdir()
    {
        Console.WriteLine(dersAdi + " Öğrenci Listesi:");
        foreach (Ogrenci ogrenci in ogrenciListesi)
        {
            Console.WriteLine(ogrenci.getAdSoyad());
        }
    }

    public string HarfNotu(double ortalama)
    {
        if (ortalama >= 90)
        {
            return "AA";
        }
        else if (ortalama >= 80)
        {
            return "BA";
        }
        else if (ortalama >= 70)
        {
            return "BB";
        }
        else if (ortalama >= 60)
        {
            return "CB";
        }
        else if (ortalama >= 50)
        {
            return "CC";
        }
        else if (ortalama >= 40)
        {
            return "DC";
        }
        else if (ortalama >= 30)
        {
            return "DD";
        }
        else
        {
            return "FF";
        }
    }
}

}